### PR TITLE
fix: Renew connection when Connection reset error occurs

### PR DESCRIPTION
**Pull request by lukaszsocha2**
Closes: SDK-2732
Fixes: #770, #765, #766

### CODE REVIEW

File: test/unit/session/test_session.py [hunk 1]
```diff
@@ -221,6 +222,42 @@ def test_box_session_raises_for_failed_response(box_session, mock_network_layer,
...
+def test_box_session_retries_requests_library_exceptions(box_session, mock_network_layer, generic_successful_request_response, test_url):
+    mock_network_layer.request.side_effect = [RequestException(), generic_successful_request_response]
+    mock_network_layer.retry_after.side_effect = lambda delay, request, *args, **kwargs: request(*args, **kwargs)
+
+    box_response = box_session.get(url=test_url)
+    assert box_response.status_code == 200
+
+
+def test_box_session_raises_requests_library_exception_when_max_retries_limit_is_reached(box_session, mock_network_layer, test_url):
+    mock_network_layer.request.side_effect = [RequestException()] * 6
```
Comment (antusus) [inline test/unit/session/test_session.py:234]: Don't you want to use API.MAX_RETRY_ATTEMPTS + 1 instead of hardcoded 6?

**Review by antusus — APPROVED** [review]
👍

File: test/unit/session/test_session.py (hunk 1 above)
Comment (lukaszsocha2) [reply test/unit/session/test_session.py:234]: makes sense

**Review by antusus — APPROVED** [review]
👍

**Pull request merged**